ui/src/shared/components/ResizeContainer.js

```diff
@@ -98,6 +100,11 @@ const ResizeContainer = React.createClass({
     const {topHeight, bottomHeight, isDragging} = this.state
     const {containerClass, children} = this.props
 
+    if (children.length > maximumNumChildren) {
```

Comment:
have you tested this? there are special `React.Children` methods meant for stuff like this